### PR TITLE
dev-cpp/gtest: Fix test failures

### PR DESCRIPTION
**Pull request by Peter-Levine**
Having tested with qemu, I can confirm that `gmock-matchers-misc_test` fails with an arm-hardfp stage3 as hinted by [Debian](https://salsa.debian.org/debian/googletest/-/blob/master/debian/rules?ref_type=heads#L7) , but not with the non-hardfp arm stage3.   Replacing `-O*` with `-O1` works around it.

I've also confirmed that `googletest-printers-test`  fails in an x86 CHOST chroot due to a floating-point precision discrepancy. As hinted by [Debian](https://salsa.debian.org/debian/googletest/-/commit/8ae6af68d69aa9cd287fdfa7f86f240483d074e1), adding `-ffloat-store` works around it.

`gtest-1.14.0` was also successfully tested using `dev-lang/python-p3.13.0_beta1_p3:3.13::gentoo` and the `3.13` slot was added to `PYTHON_COMPAT `.

- **dev-cpp/gtest: enable py3.13**

1. **dev-cpp/gtest: fix test failure with -O2/-O3 on arm hardfloat arch**
Testing on arm hardfloat architecture with -O2/-O3 causes a unittest to
fail.  Do like Debian does and replace with -O1 until upstream
addresses it.

- **dev-cpp/gtest: fix test failure on x86**
Testing on x86 architecture causes a unittest to fail due to
a floating-point precision discrepancy.  Like Debian, we add
-ffloat-store until upstream addresses the issue.

Closes: https://bugs.gentoo.org/905007
Closes: https://bugs.gentoo.org/925093
Suggested-by: Sam James <sam@gentoo.org>
Signed-off-by: Peter Levine <plevine457@gmail.com>

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by thesamesam** [comment]
Thank you! I find the ARM fix odd and a reduced version we could report to GCC (if it's indeed not UB, anyway) would be great, but I can't insist you do that for something the size of gtest.

File: dev-cpp/gtest/gtest-1.14.0.ebuild [hunk 1]
```diff
@@ -47,7 +47,14 @@ multilib_src_configure() {
 		-Dgmock_build_tests=$(usex test)
 		-Dgtest_build_tests=$(usex test)
 	)
-	use test && mycmakeargs+=( -DPython3_EXECUTABLE="${PYTHON}" )
+	if use test; then
+		if use arm && [[ $(tc-is-softfloat) =~ (softfp)|(no) ]]; then
+			replace-flags -O* -O1 # bug #925093
```
Comment (thesamesam) [inline dev-cpp/gtest/gtest-1.14.0.ebuild:52]: Why only for tests? Are we sure it's a test-only issue in the arm case?

For the x86 one, I can accept it's likely a test-only issue though.

Comment (Peter-Levine) [reply dev-cpp/gtest/gtest-1.14.0.ebuild:52]: OK. Then a revbump would be in order.

Comment (thesamesam) [reply dev-cpp/gtest/gtest-1.14.0.ebuild:52]: Yes please.

**Comment by thesamesam** [comment]
Thanks for looking. I'll keep it in the back of my head to look at.

**Comment by thesamesam** [comment]
Thanks! Let's file a stablereq for it immediately / update the existing one.

**Pull request closed without merging**